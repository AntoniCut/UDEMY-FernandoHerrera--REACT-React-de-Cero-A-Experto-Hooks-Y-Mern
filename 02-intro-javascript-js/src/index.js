//  *************************************
//  **********  /src/index.js  **********
//  *************************************


import './bases/01-const-let';
import './bases/02-template-string';
import './bases/03-objetos-literales';
import './bases/04-arreglos';
import './bases/05-funciones';
import './bases/06-desestructuracion-objetos';
import './bases/07-desestructuracion-arreglos';
import './bases/08-import-export';
import './bases/09-promesas';
import './bases/10-fetch-api';





