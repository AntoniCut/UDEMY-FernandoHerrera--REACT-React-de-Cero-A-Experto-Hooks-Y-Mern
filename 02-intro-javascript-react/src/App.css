/*  ************************************  */
/*  **********  /src/App.css  **********  */
/*  ************************************  */


.App {
  text-align: center;
  box-sizing: border-box;
  border: 10px solid #61dafb;
}

.App-logo {
  height: 20vmin;
  pointer-events: none;
  width: 200px;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 15s linear;
  }
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}



.App-header {
  max-width: 100%;
  background-color: #282c34;
  min-height: 100px; 
  display: flex;
  flex-direction: row;
  /* align-items: center; */
  justify-content: center;
  justify-content: space-around;
  font-size: calc(10px + 2vmin);
  color: white;
  border-top: 10px solid;
  border-top-color: #61dafb;
}

.App-content {
  max-width: 100%;
  background-color: #282c34 ;
  border-top: 10px solid;
  border-top-color: #61dafb;
}

.App-link {
  color: #61dafb;
}

.content-component {
  display: flex;
  flex-direction: column;
  align-items: left;
  margin-left: 300px;
}

.resultado {
  display: flex;
  width: auto;
  justify-content: left;
}

.title {
  font-size: 50px;
  color: #ffffff;
}

.text {
  font-size: 16px;
  color: #ffffff;
  text-align: left;
}

.contenido {
  display: flex;
  flex-direction: column;
  /* align-items:self-start; */
  width: 300px;
}

h3 {
  margin-left: 70px;
  display: flex;
  width: 1000px;
  /* width: 60%; */
  /* justify-content: space-between; */
}

.label {
  width: 400px;
}

.result {
  /* width: 500px;  */
  /* width: 1000px;    */
  /* width: auto; */
  /* width: 100%; */
  font-size: 20px;
  margin-left: 50px;
  text-align: left;
  color: bisque;
}

.linea {
  display: flex;
  width: 100%;
  height: 5px;
  margin-top: 50px;
  /* margin-bottom: 30px; */
  background-color: #61dafb;
}


/*  **********  Responsive  **********  */

@media (max-width: 1000px) {

  .content-component {
    display: flex;
    flex-direction: column;
    align-items: left;
    margin-left: 0;
  }
  
  .text {
    text-align: center;
  }
  
  .title {
    font-size: 30px;
  }

}


@media (max-width: 725px) {

  h3 {
    margin-left: 0;
    flex-direction: column;
    width: auto;
    justify-content: center;
    align-items: center;
  }

  .contenido {
    display: flex;
    flex-direction: column;
    align-items: center;
    width: 300px;
  }

  .resultado {
    display: flex;
    width: auto;
    justify-content: center;
  }

  .result {
    text-align: center;
    font-size: 14px;
  }

}


@media (max-width: 600px) {

  .App-header {
    flex-direction: column;
    align-items: center; 
    justify-content: center;
  }

}