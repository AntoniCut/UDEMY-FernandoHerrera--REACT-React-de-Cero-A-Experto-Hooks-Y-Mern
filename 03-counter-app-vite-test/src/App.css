/*  ********************************************************  */
/*  **********  /03-counter-app-vite/src/App.css  **********  */
/*  ********************************************************  */


* {
    margin: 0;
    padding: 0;
    box-sizing: border-box; 
}

/* hr {
    width: 100%;
    height: 2px;
    margin: 20px 0;
} */

.main {
    width: 100%;
    min-height: 100vh; 
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    background-color: bisque;
}

.header {
    display: flex;
    /* flex-wrap: wrap; */
    justify-content: center;
    align-items: center;
    width: 100%; 
    min-height: 150px;
    gap: 100px;
    background-color: rgb(202, 212, 212);
}

.card {
    display: flex;
    flex-direction: column;
    /* justify-content: center;  */
    align-items: center;
    width: 100%;
    margin-top: 50px;
    /* flex-grow: 1; */
    background-color: bisque;
}

.logo {
    height: 10em;
    padding: 1.5em;
    will-change: filter;
    transition: filter 300ms;
}

.logo:hover {
    filter: drop-shadow(0 0 2em #646cffaa);
}

.logo.react:hover {
    filter: drop-shadow(0 0 2em #61dafbaa);
}

@keyframes logo-spin {
    from {
        transform: rotate(0deg);
    }

    to {
        transform: rotate(360deg);
    }
}

@media (prefers-reduced-motion: no-preference) {
    a:nth-of-type(2) .logo {
        animation: logo-spin infinite 15s linear;
    }
}

.titulo {
    color: #000;
}

.subtitulos {
    color: #4cb4b4;
}

.resultados {
    color: #5BB0FF;
}

code {
    font-size: 24px;
}



.counter-container {
    display: flex;
    flex-direction: column;
}

.counter-superior {
    display: flex;
    flex-direction: row;
    gap: 50px;
}

.counter-inferior {
    display: flex;
    justify-content: center;
    margin-top: 40px;
}

.btn-counter {
    height: 40px;
    line-height: 0px;
    text-align: center;
    font-size: 24px;
    transition: all 300ms ease-in-out;
}

.btn-counter:hover {
    background-color: #5BB0FF;
}


.counter-marcador {
    width: 80px;
    height: 80px;
    text-align: center;
    background-color: white;
    border-radius: 100%;

}

.counter-font-size {
    font-size: 48px;
}

.margen-inferior {
    margin-bottom: 50px;
}


/*  **********  Responsive  **********  */

@media( width < 1250px ) {

    .header {
        background-color: #4cb4b4;
        gap: 50px;
    }

    .header h1 {
        font-size: 36px;
        /* line-height: 1.1; */
      }

}


@media( width < 1000px ) {

    .logo {
        height: 8em;
    }

    .header h1 {
        font-size: 24px;
        gap: 30px;
        /* line-height: 1.1; */
      }

}





