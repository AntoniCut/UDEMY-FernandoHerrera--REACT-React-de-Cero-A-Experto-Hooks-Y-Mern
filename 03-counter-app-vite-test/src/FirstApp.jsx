//  *************************************************************
//  **********  /03-counter-app-vite/src/FirstApp.jsx  **********
//  *************************************************************

import PropTypes from "prop-types";


//  -----  Definicion de Variables, Objetos y Funciones  -----
const newMessage = 'Antonio';
const arr = [1, 2, 3, 4, 5, 6, 7, 8, 9];
const obj = {
    message: 'Hola Mundo',
    title: 'Antonio'
}

const misValores = () => {

    const newMessage = 'Antonio';
    const arr = [1, 2, 3, 4, 5, 6, 7, 8, 9];
    const obj = {
        message: 'Hola Mundo',
        title: 'Antonio'
    }

    const resultado = [newMessage, arr, obj.message, obj.title];
    return resultado.join(', ');
}



//  **********  Componente FirstApp()  **********
export const FirstApp = ({ title1, subTitle, cantidades, name }) => {

    console.log({title1});
    console.log({subTitle});
    console.log({cantidades});

    //if(!title1) throw new Error('El title no existe');
        
    
    return (
        <>
            <h1 className="titulo"> First App  </h1>
            <h2> del Curso de React </h2>
            <br />

            <h2 className="subtitulos"> Titulo: <span className="resultados"> {title1} </span> </h2> 
            <h2 className="subtitulos"> Subtitulo: <span className="resultados"> {subTitle} </span> </h2> 
            <h2 className="subtitulos"> Cantidades: <span className="resultados"> {cantidades} </span> </h2>
            <h2 className="subtitulos"> Name:  <span className="resultados"> {name} </span> </h2>

            <br />

            <h3 className="subtitulos"> La Función misValores() devuelve - <span className="resultados"> {misValores()} </span> </h3>

            <h3 className="subtitulos"> Mi nombre es: <span className="resultados"> {newMessage} </span> </h3>
            <h4 className="subtitulos"> Arr: <span className="resultados"> {arr} </span> </h4> <br />
            <h4 className="subtitulos"> Obj: <span className="resultados"> {obj.message} - {obj.title} </span> </h4>

            {/* Serializa el Objeto */}
            <code className="resultados"> {JSON.stringify(obj)} </code>
        </>
    );
}


//  -----  Definición de las Props  -----
FirstApp.propTypes = {

    title1: PropTypes.string.isRequired,
    subTitle: PropTypes.string.isRequired,
    cantidades: PropTypes.number.isRequired,
    name: PropTypes.string
}


//  -----  Default Props  -----

// FirstApp.defaultProps = {
//     title1: 'No Hay Título',
//     subTitle: 'No hay Subtitulo',
//     cantidades: 0,
//     name: 'Antonio Cutillas'
// }