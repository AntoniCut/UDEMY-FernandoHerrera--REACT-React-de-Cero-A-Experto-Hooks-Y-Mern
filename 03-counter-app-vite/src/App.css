/*  ********************************************************  */
/*  **********  /03-counter-app-vite/src/App.css  **********  */
/*  ********************************************************  */


* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
}

hr {
  width: 100%;
  height: 2px;
  margin: 20px 0;
}

.main {
  display: flex;
  flex-direction: column;
  min-width: 1920px;
  width: 100%;
  min-height: 100vh;
  background-color: bisque;
}

.header {
  display: flex;
  justify-content: center;
  /* Espacio uniforme entre elementos */
  align-items: center;
  width: 100%;
  /* Ocupar todo el ancho */
  height: 150px;
  gap: 100px;
  background-color: rgb(202, 212, 212);
}

.card {
  display: flex;
  flex-direction: column;
  /* justify-content: center;  */
  align-items: center;
  width: 100%;
  margin-top: 50px;
  flex-grow: 1;
  background-color: bisque;
}

.logo {
  height: 10em;
  padding: 1.5em;
  will-change: filter;
  transition: filter 300ms;
}

.logo:hover {
  filter: drop-shadow(0 0 2em #646cffaa);
}

.logo.react:hover {
  filter: drop-shadow(0 0 2em #61dafbaa);
}

@keyframes logo-spin {
  from {
    transform: rotate(0deg);
  }

  to {
    transform: rotate(360deg);
  }
}

@media (prefers-reduced-motion: no-preference) {
  a:nth-of-type(2) .logo {
    animation: logo-spin infinite 15s linear;
  }
}

.titulo {
  color: #000;
}

.subtitulos {
  color: #4cb4b4;
}

.resultados {
  color: #5BB0FF;
}

code {
  font-size: 24px;
}



.counter-container {
  display: flex;
  flex-direction: column;
}

.counter-superior {
  display: flex;
  flex-direction: row;
  gap: 50px;
}

.counter-inferior {
  display: flex;
  justify-content: center;
  margin-top: 40px;
}

.btn-counter {
  height: 40px;
  line-height: 0px;
  text-align: center;
  font-size: 24px;
  transition: all 300ms ease-in-out;
}

.btn-counter:hover {
  background-color: #5BB0FF;
}


.counter-marcador {
  width: 80px;
  height: 80px;
  text-align: center;
  background-color: white;
  border-radius: 100%;

}

.counter-font-size {
  font-size: 48px;
}

.margen-inferior {
  margin-bottom: 50px;
}