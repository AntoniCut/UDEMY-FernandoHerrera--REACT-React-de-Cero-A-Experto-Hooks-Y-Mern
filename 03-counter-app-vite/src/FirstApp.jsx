//  ********************************************************
//  **********  /03-counter-app-vite/src/App.jsx  **********
//  ********************************************************


const newMessage = 'Antonio';
const arr = [1, 2, 3, 4, 5, 6, 7, 8, 9];
const obj = {
    message: 'Hola Mundo',
    title: 'Antonio'
}


const misValores = () => {
    
    const newMessage = 'Antonio';
    const arr = [1, 2, 3, 4, 5, 6, 7, 8, 9];
    const obj = {
        message: 'Hola Mundo',
        title: 'Antonio'
    }

    const resultado = [newMessage, arr, obj.message, obj.title];
    return resultado.join(', ');
}



//  **********  Componente FirstApp()  **********
export const FirstApp = () => {

    return (
        <>
            <h1> First App </h1> <br />
            <h2> del Curso de React </h2> <br />
            
            <h3> La Funci√≥n misValores() devuelve - {misValores()}</h3>

            <h3> Mi nombre es: {newMessage} </h3> 
            <h4> {arr} </h4> <br />
            <h4> {obj.message} - {obj.title} </h4>
            
            {/* Serializa el Objeto */}
            <code> {JSON.stringify(obj)} </code>
        </>
    );
}

