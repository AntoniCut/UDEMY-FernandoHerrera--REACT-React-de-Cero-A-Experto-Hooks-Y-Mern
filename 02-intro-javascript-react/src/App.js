//  ***********************************
//  **********  /src/App.js  **********
//  ***********************************


import logo from './logo.svg';
import './App.css';
import { ConstLetComponent } from './bases/01-const-let';
import { TemplateStringComponent } from './bases/02-template-string';
import { ObjetosLiteralesComponent } from './bases/03-objetos-literales';
import { ArreglosComponent } from './bases/04-arreglos';
import { FuncionesComponent } from './bases/05-funciones';
import { DesestructuracionObjetosComponent } from './bases/06-desestructuracion-objetos';
import { DesestructuracionArreglosComponent } from './bases/07-desestructuracion-arreglos';
import { ImportExportComponent } from './bases/08-import-export';
import { PromesasComponent } from './bases/09-promesas';
import { FetchApisComponent } from './bases/10-fetch-apis';
import { CondicionalTernarioComponent } from './bases/12-condicional-ternario';




//  **********  Componente Principal *****  App()  **********
const App = () => {


    return (

        <div className="App">

            <header className="App-header">
                <img src={logo} className="App-logo" alt="logo" />
                <h1> 02 Introducci√≥n de JavaScript </h1>
                <img src={logo} className="App-logo" alt="logo" />
            </header>

            <main className="App-content">

                <div className='content-component'>
                    <div className='resultado'>
                        <ConstLetComponent />
                    </div>
                </div>
                <div className='linea'> </div>


                <div className='content-component'>
                    <div className='resultado'>
                        <TemplateStringComponent />
                    </div>
                </div>
                <div className='linea'> </div>


                <div className='content-component'>
                    <div className='resultado'>
                        <ObjetosLiteralesComponent />
                    </div>
                </div>
                <div className='linea'> </div>


                <div className='content-component'>
                    <div className='resultado'>
                        <ArreglosComponent />
                    </div>
                </div>
                <div className='linea'> </div>


                <div className='content-component'>
                    <div className='resultado'>
                        <FuncionesComponent />
                    </div>
                </div>
                <div className='linea'> </div>


                <div className='content-component'>
                    <div className='resultado'>
                        <DesestructuracionObjetosComponent />
                    </div>
                </div>
                <div className='linea'> </div>


                <div className='content-component'>
                    <div className='resultado'>
                        <DesestructuracionArreglosComponent />
                    </div>
                </div>
                <div className='linea'> </div>


                <div className='content-component'>
                    <div className='resultado'>
                        <ImportExportComponent />
                    </div>
                </div>
                <div className='linea'> </div>


                <div className='content-component'>
                    <div className='resultado'>
                        {/* <PromesasComponent /> */}
                    </div>
                </div>
                <div className='linea'> </div>


                <div className='content-component'>
                    <div className='resultado'>
                        {/* <FetchApisComponent /> */}
                    </div>
                </div>
                <div className='linea'> </div>


                <div className='content-component'>
                    <div className='resultado'>
                        <CondicionalTernarioComponent /> 
                    </div>
                </div>
                <div className='linea'> </div>


            </main>

        </div>

    );
}

export default App;



