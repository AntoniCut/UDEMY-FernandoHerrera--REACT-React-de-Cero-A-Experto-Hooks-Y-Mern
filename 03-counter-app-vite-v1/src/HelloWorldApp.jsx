//  ******************************************************************
//  **********  /03-counter-app-vite/src/HelloWorldApp.jsx  **********
//  ******************************************************************


import { useState } from 'react'
import reactLogo from './assets/react.svg'
import viteLogo from '/vite.svg'
import './App.css'
import { FirstApp } from './FirstApp'
import { CounterApp } from './CounterApp'


//  **********  Componente Principal HelloWorldApp()  **********
export const HelloWorldApp = () => {


    const [count, setCount] = useState(0)


    return (

        <main className='main'>

            <div className='header'>

                <a href="https://vitejs.dev" target="_blank">
                    <img src={viteLogo} className="logo" alt="Vite logo" />
                </a>

                <h1> Hello World App !!! </h1>

                <button onClick={() => setCount((count) => count + 1)}>
                    count is {count}
                </button>

                <a href="https://react.dev" target="_blank">
                    <img src={reactLogo} className="logo react" alt="React logo" />
                </a>

            </div>


            <div className="card">

                <FirstApp title1="My Primer Componente de React" subTitle="Subtitulo del Componente" cantidades={123} />
                {/* <FirstApp  /> */}

                <div className="margen-inferior"> </div>

                <CounterApp value={-100} />
                {/* <CounterApp /> */}

                <div className="margen-inferior"> </div>

            </div>

        </main>
    );
}



