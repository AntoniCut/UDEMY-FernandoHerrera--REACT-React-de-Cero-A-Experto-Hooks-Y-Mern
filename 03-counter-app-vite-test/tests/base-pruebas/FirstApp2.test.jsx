//  ********************************************************************
//  **********  /03-counter-app-vite/test/FirstApp2.test.jsx  **********
//  ********************************************************************


import { render, screen } from '@testing-library/react';
import { FirstApp } from '../../src/FirstApp';


describe('Pruebas en <FirstApp />', () => {

    const title = 'Hola, Soy Goku';
    const subTitle = 'Soy un Subtitulo';


    test('debe de hacer match con el snapshot', () => {

        const { container } = render(<FirstApp title={title} />);

        expect(container).toMatchSnapshot();

    });


    test('debe de mostrarel mensaje "Hola, Soy Goku', () => {

        //  -----  es donde se renderiza el componente  -----
        render(<FirstApp title1={title} />);

        //screen.debug();

        expect(screen.getByText(title)).toBeTruthy();

    });


    test('debe de mostrar el titulo en un h1', () => {

        render(<FirstApp title1={title} />);

        expect( screen.getByRole('heading', {level: 1} ).innerHTML ).toContain( title) ; 
        
    });


    test('debe de mostrar el subtitulo enviado por props', () => {

               
        render(
            <FirstApp
                title1={title}
                subTitle={subTitle}
                
            />
        );

        expect( screen.getAllByText(subTitle).length).toBe(2);  //  Tenemos 2 subtitulos.

    });


});