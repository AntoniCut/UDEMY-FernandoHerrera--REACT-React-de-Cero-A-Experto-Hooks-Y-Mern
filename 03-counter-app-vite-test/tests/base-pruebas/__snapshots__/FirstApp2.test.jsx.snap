// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp /> debe de hacer match con el snapshot 1`] = `
<div>
  <h2
    class="titulo"
  >
     First App  
  </h2>
  <h2>
     del Curso de React 
  </h2>
  <br />
  <h1
    class="subtitulos"
    data-testid="test-title"
  >
     Titulo: 
    <span
      class="resultados"
    >
       
       
    </span>
     
  </h1>
  <h2
    class="subtitulos"
  >
     Subtitulo: 
    <span
      class="resultados"
    >
       
      No hay Subtitulo
       
    </span>
     
  </h2>
  <h2
    class="subtitulos"
  >
     Subtitulo: 
    <span
      class="resultados"
    >
       
      No hay Subtitulo
       
    </span>
     
  </h2>
  <h2
    class="subtitulos"
  >
     Cantidades: 
    <span
      class="resultados"
    >
       
      0
       
    </span>
     
  </h2>
  <h2
    class="subtitulos"
  >
     Name:  
    <span
      class="resultados"
    >
       
      Antonio Cutillas
       
    </span>
     
  </h2>
  <br />
  <h3
    class="subtitulos"
  >
     La Funci√≥n misValores() devuelve - 
    <span
      class="resultados"
    >
       
      Antonio, 1,2,3,4,5,6,7,8,9, Hola Mundo, Antonio
       
    </span>
     
  </h3>
  <h3
    class="subtitulos"
  >
     Mi nombre es: 
    <span
      class="resultados"
    >
       
      Antonio
       
    </span>
     
  </h3>
  <h4
    class="subtitulos"
  >
     Arr: 
    <span
      class="resultados"
    >
       
      1
      2
      3
      4
      5
      6
      7
      8
      9
       
    </span>
     
  </h4>
   
  <br />
  <h4
    class="subtitulos"
  >
     Obj: 
    <span
      class="resultados"
    >
       
      Hola Mundo
       - 
      Antonio
       
    </span>
     
  </h4>
  <code
    class="resultados"
  >
     
    {"message":"Hola Mundo","title":"Antonio"}
     
  </code>
</div>
`;
